store/userAction.ts

```diff
@@ -0,0 +1,20 @@
+import { createAction, createAsyncAction } from 'typesafe-actions';
+import { AxiosError } from 'axios';
+
+type UserResponse = {
+  birthday: string;
+  email: string;
+  name: string;
+};
+
+export const SET_USER_LOADING = 'user/set-user-loading' as const;
+export const setUserLoading = createAction(SET_USER_LOADING)<boolean>();
+
+export const GET_USER_NAME_REQUEST = 'user/get-user-name-request' as const;
+export const GET_USER_NAME_SUCCESS = 'user/get-user-name-success' as const;
+export const GET_USER_NAME_FAILURE = 'user/get-user-name-failure' as const;
```

Comment:
const로 선언해서 as const는 필요 없지 않아?
```suggestion
export const SET_USER_LOADING = 'user/set-user-loading';
export const setUserLoading = createAction(SET_USER_LOADING)<boolean>();

export const GET_USER_NAME_REQUEST = 'user/get-user-name-request';
export const GET_USER_NAME_SUCCESS = 'user/get-user-name-success';
export const GET_USER_NAME_FAILURE = 'user/get-user-name-failure';
```

Reply:
readonly 만드려고 const assertions 쓴건데 그냥 문자열에서는 빼도 똑같은듯 빼겟슴다